be2-scala/src/main/scala/ch/epfl/pop/authentication/PopchaWebSocketResponseHandler.scala

```diff
@@ -0,0 +1,82 @@
+package ch.epfl.pop.authentication
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.http.scaladsl.model.ws.{Message, TextMessage}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+import akka.stream.scaladsl.{Flow, Sink, Source}
+import akka.stream.{CompletionStrategy, OverflowStrategy}
+import akka.{Done, NotUsed}
+import ch.epfl.pop.config.ServerConf
+
+import scala.concurrent.{Future, Promise}
+
+/** Websocket handler that, given two websocket connections on a common endpoint, forwards a single message from the second websocket connected to the first:
+  *   - Connection requests are expected to lend on endpoints with format "/laoId/authentication/clientId/nonce". This uniquely determines a connection by its socketId (laoId, clientId, nonce).
+  *   - The first websocket connected on a given endpoint is assumed to be waiting for a message. It will be provided by the second websocket to connect on that same endpoint (at any point later in time).
+  *   - Once a message is received from the second websocket connected on a given endpoint, it is forwarded to the first websocket connected and both connections are then closed.
+  */
```

Comment:
I think the code speaks for itself and the doc should just give a higher level explanation, what do you think ?

```suggestion
/** This websocket handler receives the JWT from the PopchaHandler via websocket and forwards it to the client webpage connected via websocket as well.
  */
```

Reply:
Yes that makes sense !